tcconfig/shaper/htb.py

```diff
@@ -112,10 +132,25 @@ def _add_rate(self):
         ]
 
         if bandwidth != upper_limit_rate:
+            # This sets the burst/cburst values to the default tc tries to set,
+            # but works around a bug in tc where values are rounded incorrectly.
+            
+            # Note that "tc class show" will still show burst that are too small,
+            # due to the inverse bug, but if you use "tc -r class show" you will see
+            # the actual size of the cbuffer in bytes.
+            mtu = 1600
+            rate = bandwidth.byte_per_sec
+            desired_burst = int(rate / get_hz() + mtu)
```

Comment:
Is the reason for adding `mtu` in this calculation to provide a minimum guaranteed value?
If so, the following seems appropriate:

```suggestion
            desired_burst = max(int(rate / get_hz()), mtu)
```

Reply:
This is the default burst size calculation that tc tries to set, see [line](https://git.kernel.org/pub/scm/network/iproute2/iproute2.git/tree/tc/q_htb.c#n226).   (Unfortunately the value it sets is then mangled by tc's buggy calc_xmittime implementation.)

The comment there is
```
        /* compute minimal allowed burst from rate; mtu is added here to make
           sure that buffer is larger than mtu and to have some safeguard space */
```

so I followed what they implemented.

Reply:
Thank you for your explanations.
Could you add those insights as comments?